app/master/build.py

```diff
@@ -326,6 +341,11 @@ def cancel(self):
         self._state_machine.trigger(BuildEvent.CANCEL)
 
     def _on_enter_canceled_state(self, event):
+        # Set the kill_event to kill the subprocesses for the build, if build cancel is called during PREPARING state
+        if event.src == BuildState.PREPARING:
+            self._logger.notice('Set the kill_event for subprocesses for build {}.', self._build_id)
```

Comment:
I agree with `omitting the check for BuildState.PREPARING`. Its simpler, and will work when in future if we add any subprocess to other BuildStates.